File: 3371-harshad-number/3371-harshad-number.cpp
class Solution {
public:
    int sumOfTheDigitsOfHarshadNumber(int x) {
        int c=0;
        int r=x;
        while(x){
            int rem=x%10;
            c+=rem;
            x/=10;
        }
        if(r%c==0){
            return c;
        }
        return -1;
    }
};

File: 3371-harshad-number/3371-harshad-number.py
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        sum=0
        r=x
        while(x):
            rem=x%10
            sum+=rem
            x//=10
        if(r%sum==0):
            return sum
        return -1
        